drivers/adc/adc_stm32.c

```diff
@@ -263,9 +239,11 @@ struct adc_stm32_cfg {
 	void (*irq_cfg_func)(void);
 	struct stm32_pclken pclken;
 	const struct pinctrl_dev_config *pcfg;
-	bool has_temp_channel;
-	bool has_vref_channel;
-	bool has_vbat_channel;
+	int8_t temp_channel;
+	int8_t vref_channel;
+	int8_t vbat_channel;
+	int8_t res_table_size;
+	const int32_t *res_table;
```

Comment:
It looks like the values were small enough to use a `uint16_t` here.

Reply:
Fixed